src/main/scala/interfaces/Converter.scala

```diff
@@ -0,0 +1,318 @@
+            buf += "\t" + name + " <- " + entry.toString.split("\n").mkString("\n\t") + "\n"
+        }
+        buf += "}"
+        return buf
+    }
+}
+case class VarEntry(name: String) {
+    override def toString = name
+}
+case class NullRefEntry() extends ExtractedModelEntry {
+    override def toString = "Null()"
+}
+case class OtherEntry(value: String) extends ExtractedModelEntry {
+    override def toString = value + "(unhandled type)"
+}
+
+case class ExtractedModel(entries: Map[String, ExtractedModelEntry]) {
+    override def toString : String = {
+        var buf = ""
+        for ((name, entry) <- entries) {
+            buf += name + " <- " + entry.toString + "\n"
+        }
+        return buf
+    }
+}
+
+sealed trait HeapEntry
+case class PredHeapEntry(name: String, args: Array[Term]) extends HeapEntry {
+    override def toString = name + "(" + args.mkString(", ") +")"
+}
+case class FieldHeapEntry(recv: String, field: String, perm: Rational, typ: Option[ast.Type]) extends HeapEntry {
+    override def toString = recv + "." + field
+}
+
+case class ExtractedHeap(entries: Map[HeapEntry, String]) {
+    override def toString = entries.map(x => x._1.toString + " <- " + x._2).mkString("\n")
+}
+
+
+/* basically a 1 to 1 copy of nagini code */
+object Converter{    
+    type ExtractedHeap = Map[HeapEntry, String]
+    def snapToOneLine(s:String) : String = s.filter(_ >= ' ').split(" +").mkString(" ")
+
+    def getParts(value: String) : Array[String] = {
+        var res : Array[String] = Array()
+        for (part <- ModelParser.getApplication(value)) {
+            res = res :+ part
+        }
+        return res
+    }
+
+    def getFunctionValue(model: Model, fname: String, args: String) : String = {
+        val entry : ModelEntry = model.entries(fname)
+        entry match {
+            case SingleEntry(s) => return s 
+            case MapEntry(m: Map[Seq[String], String], els:String) =>
+                val filtered = m.filter(x => snapToOneLine(x._1.mkString(" ")) == args)
+                if (filtered.size >= 1) {
+                    return filtered.head._2.toString
+                }
+                return els
+        }
+    }
+
+
+    def translateSort(s:Sort) : String = {
+        s match {
+            case sorts.Set(els:Sort) => return "Set<" + translateSort(els)+">"
+            case sorts.Ref => return "$Ref"
+            case sorts.Snap => return "$Snap"
+            case sorts.Perm => return "$Perm"
+            case sorts.Seq(els:Sort) => return "Seq<" + translateSort(els)+">"
+            case _ => s.toString
+        }
+    }
+
+    def sortToType(s:Sort) : Option[ast.Type] = {
+        s match {
+            case sorts.Ref => return Some(ast.Ref)
+            case sorts.Int => return Some(ast.Int)
+            case sorts.Bool => return Some(ast.Bool)
+            case sorts.Perm => return Some(ast.Perm)
+            case _ => println("This type of Heap Field can not be handled : " + s.toString)
```

Comment:
`println`ing an error is most likely not a good idea. If you want debug output, use Silicon's logger(s) for that. If you want an error, raise an exception.

Reply:
I am a bit confused by "silicon's logger". Is it okay if I use the logger from com.typesafe.scalalogging.Logger? I saw it's used in other Files too.
Or what exactly should I use?